crud/src/main/java/com/redhat/lightblue/assoc/ep/ExecutionPlan.java

```diff
@@ -0,0 +1,474 @@
+                        search.setLimit(from,to);
+                        search.setSort(requestSort);
+                        sorted=limited=true;
+                    } else {
+                        // There are unassigned clauses. We can sort during search, but we have to filter and limit
+                        search.setSort(requestSort);
+                        sorted=true;
+                        Filter f=new Filter(block,new Source<>(search),getFilterQuery(unassigned));
+                        last=new Source<>(f);
+                        if(from!=null)
+                            last=new Source<>(new Skip(block,from.intValue(),last));
+                        if(to!=null)
+                            last=new Source<>(new Limit(block,to.intValue()-from.intValue()+1,last));
+                        limited=true;
+                    }
+                }
+                search.recordResultSetSize(true);
+                
+                Set<Path> fields=getIncludedFieldsOfEntityForSearch(block,qfi);
+                fields.addAll(getIncludedFieldsOfEntityForProjection(block,rootMd,requestProjection));
+                search.setProjection(writeProjection(fields));
+                search.setQueries(node.getData().getConjuncts());
+                resultStep=new Assemble(block,last,destinationBlocks);
+                resultStep=new Project(block,new Source<>(resultStep),requestProjection);
+                block.setResultStep(resultStep);
+            } else {
+                // Processing one of the associated entity nodes
+                // Reuse the batching algorithm in join                
+                ExecutionBlock source=block.getSourceBlocks().get(0);
+                QueryPlanData edgeData=retrievalQueryPlan.getEdgeData(source.getQueryPlanNode(),
+                                                                      block.getQueryPlanNode());
+                if(edgeData!=null) {
+                    AssociationQuery aq=new AssociationQuery(rootMd,
+                                                             block.getMetadata(),
+                                                             block.getReference(),
+                                                             edgeData.getConjuncts());
+                    block.setAssociationQuery(source,aq);
+                }
+                Retrieve search=new Retrieve(block);
+                search.setQueries(node.getData().getConjuncts());
+                Set<Path> fields=getIncludedFieldsOfEntityForSearch(block,qfi);
+                fields.addAll(getIncludedFieldsOfEntityForProjection(block,rootMd,requestProjection));
+                search.setProjection(writeProjection(fields));
+                block.setResultStep(new Assemble(block,new Source<>(search),destinationBlocks));
+            }
+        }
+
+        for(ExecutionBlock block:qp2BlockMap.values()) {
+            block.initialize();
+        }
+    }
```

Comment:
I know, and this was the best I could do without writing functions with
ugly parameter lists. Any suggestions on how to do it?

On Tue, Mar 29, 2016 at 4:34 PM, Brandon Vulaj <notifications@github.com>
wrote:

> In crud/src/main/java/com/redhat/lightblue/assoc/ep/ExecutionPlan.java
> <https://github.com/lightblue-platform/lightblue-core/pull/608#discussion_r57812443>
> :
>
> > +                                                             edgeData.getConjuncts());
> > +                    block.setAssociationQuery(source,aq);
> > +                }
> > +                Retrieve search=new Retrieve(block);
> > +                search.setQueries(node.getData().getConjuncts());
> > +                Set<Path> fields=getIncludedFieldsOfEntityForSearch(block,qfi);
> > +                fields.addAll(getIncludedFieldsOfEntityForProjection(block,rootMd,requestProjection));
> > +                search.setProjection(writeProjection(fields));
> > +                block.setResultStep(new Assemble(block,new Source<>(search),destinationBlocks));
> > +            }
> > +        }
> > +
> > +        for(ExecutionBlock block:qp2BlockMap.values()) {
> > +            block.initialize();
> > +        }
> > +    }
>
> 300 line ctor - any better way to break this up just for readability sake?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/608/files/d72d06258b9b58a53294c8eb4a994145df771c4c#r57812443>
>